src/gcp/cloudsql/fbToolsAuthClient.ts

```diff
@@ -33,13 +37,21 @@ export class FBToolsAuthClient extends AuthClient {
     };
   }
   public async getAccessToken(): Promise<{ token?: string; res?: GaxiosResponse<any> }> {
-    const client = new apiv2.Client({});
-    return { token: await client.getAccessToken() };
+    // Runtime fetch of Auth singleton to prevent circular module dependencies
+    if (apiv2.accessToken) {
+      return { token: apiv2.accessToken };
+    }
+    const data = await auth.getAccessToken(apiv2.refreshToken, []);
+    return { token: data.access_token };
   }
-  public async getRequestHeaders(url?: string | undefined): Promise<Record<string, string>> {
-    const client = new apiv2.Client({
-      urlPrefix: url,
-    });
-    return client.getHeaders();
+
+  public async getRequestHeaders(): Promise<Record<string, string>> {
+    const token = await this.getAccessToken();
+    return {
+      Connection: "keep-alive",
+      "User-Agent": `FirebaseCLI/${CLI_VERSION}`,
+      "X-Client-Version": `FirebaseCLI/${CLI_VERSION}`,
+      Authorization: `Bearer ${token.token}`,
+    };
```

Comment:
For my money, maybe make these first three either a static Map that you can iterate through and return here (adding Authorization). Then you can have apiv2 iterate over the same list.

Added bonus, you could then have a "I only want these headers" type list so that if future things are added to `apiv2.StandardHeadersMap`, you could filter them out pretty easily.